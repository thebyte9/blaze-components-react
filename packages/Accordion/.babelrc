{
  "presets": [
    "@babel/preset-env",
    [
      "@babel/preset-react",
      {
        "runtime": "automatic"
      }
    ],
    "@babel/preset-typescript"
  ],
  "plugins": [
    [
      "@babel/plugin-transform-runtime",
      {
        "regenerator": false
      }
    ],
    "@babel/plugin-syntax-jsx",
    "@babel/proposal-class-properties",
    "@babel/proposal-object-rest-spread"
  ]
}
