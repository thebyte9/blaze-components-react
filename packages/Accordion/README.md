
## Description

Use the More menu list to navigate between the items.

## Usage

```js
<More isMoreMenu>
    <More.Avatar isMoreMenu>
        <span className="material-icons">...</span>
    </More.Avatar>
    <More.Content isMoreMenu>
      <a href="/">Link</a>
      <a href="/">Link</a>
      <a href="/">Link</a>
    </More.Content>
</More>  
```

## API

##### More menu can receive a number of `props` as follow:
| NAME   | TYPE | DEFAULT | 
| :---  | :---:  | :---: | 
| isMoreMenu | Boolean | false |
| isHeader | Boolean | false |
| isDropdown| Boolean | false |
| toggled | Boolean | false |
| children | single/array of nodes | empty | 






