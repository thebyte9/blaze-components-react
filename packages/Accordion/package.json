{
  "name": "@blaze-react/accordion",
  "version": "0.4.33",
  "publishConfig": {
    "access": "public"
  },
  "description": "",
  "main": "./lib/index.js",
  "scripts": {
    "build": "../../node_modules/.bin/tsc --jsx react --esModuleInterop ./src/*.tsx ./src/**/*.tsx --outDir lib ",
    "build-storybook": "build-storybook -c .storybook -o docs-build",
    "lint": "../../node_modules/.bin/tslint -c ../../tslint.json src/**/*.tsx",
    "test": "jest",
    "test:coverage": "jest --coverage",
    "pre-publish": "npm run build && npm run lint && npm run test"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@blaze-react/button": "^0.4.33"
  },
  "devDependencies": {
    "@babel/cli": "^7.4.4",
    "@babel/core": "^7.4.5",
    "@blaze-react/blaze-components-theme": "^0.4.33",
    "@types/classnames": "^2.2.8"
  },
  "peerDependencies": {
    "prop-types": "^15.7.2",
    "react": "^16.9.0",
    "react-dom": "^16.9.0"
  },
  "babel": {
    "presets": [
      "@babel/preset-env",
      "@babel/preset-react"
    ]
  },
  "gitHead": "c7cc1d26b4046b800c727feb1dd283dbd833328c"
}
