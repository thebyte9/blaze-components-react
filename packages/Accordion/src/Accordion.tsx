import React, { useEffect, useState } from 'react';
import AccordionContent from './AccordionContent';
import AccordionContentDetails from './AccordionContent/AccordionContentDetails';
import AccordionContentFooter from './AccordionContent/AccordionContentFooter';
import AccordionHeader from './AccordionHeader';
import { DOWN, FLEX, NONE, UP } from './constants';

interface IAccordionProps {
  children: [JSX.Element, JSX.Element];
  onOpen?: () => void;
  onClose?: () => void;
  isOpen?: boolean;
}

const Accordion = ({ children, isOpen, onOpen, onClose }: IAccordionProps): JSX.Element => {
  const [accordionStatus, setAccordionStatus] = useState<string>(isOpen ? FLEX : NONE);
  const [header, content]: [JSX.Element, JSX.Element] = children;
  const isActive = accordionStatus === FLEX;

  useEffect(() => {
    setAccordionStatus(isOpen ? FLEX : NONE);
  }, [isOpen]);

  useEffect(() => {
    if (accordionStatus === FLEX) {
      onOpen && onOpen();
    } else {
      onClose && onClose();
    }
  }, [accordionStatus, onClose, onOpen])

  const toggleAccordion = (): void => {
    const status = isActive ? NONE : FLEX;
    setAccordionStatus(status);
  };

  const arrowType = isActive ? UP : DOWN;

  return (
    <div className="accordion">
      <div className="accordion__header">
        {header}
        <div className="icon-button icon-button--accordion" data-testid="toggle-accordion" onClick={toggleAccordion}>
          <i className="material-icons">{`keyboard_arrow_${arrowType}`}</i>
        </div>
      </div>
      <div className="accordion__content-wrapper" style={{ display: accordionStatus }}>
        {content}
      </div>
    </div>
  );
};

Accordion.Content = AccordionContent;
Accordion.ContentDetails = AccordionContentDetails;
Accordion.ContentFooter = AccordionContentFooter;
Accordion.Header = AccordionHeader;

Accordion.defaultProps = {
  onOpen: () => ({}),
  onClose: () => ({}),
};

export default Accordion;
