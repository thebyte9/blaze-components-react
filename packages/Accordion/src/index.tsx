import React, { useEffect, useState } from "react";
import AccordionContent from "./AccordionContent";
import AccordionContentDetails from "./AccordionContent/AccordionContentDetails";
import AccordionContentFooter from "./AccordionContent/AccordionContentFooter";
import AccordionHeader from "./AccordionHeader";

interface IAccordionProps {
  children: [JSX.Element, JSX.Element];
  open?: boolean;
}

const Accordion = ({ children, open }: IAccordionProps): JSX.Element => {
  const flex = "flex";
  const none = "none";

  const [accordionStatus, setAccordionStatus] = useState<string>(
    open ? flex : none
  );

  const [header, content]: [JSX.Element, JSX.Element] = children;

  useEffect(() => {
    setAccordionStatus(open ? flex : none);
  }, [open]);

  const isActive: boolean = accordionStatus === flex;
  const toggleAccordion = (): void =>
    setAccordionStatus(isActive ? none : flex);

  return (
    <div className="accordion">
      <div className="accordion__header">
        {header}
        <div
          className="icon-button icon-button--accordion"
          onClick={toggleAccordion}
        >
          <i className="material-icons">keyboard_arrow_down</i>
        </div>
      </div>
      <div
        className="accordion__content-wrapper"
        style={{ display: accordionStatus }}
      >
        {content}
      </div>
    </div>
  );
};

Accordion.Content = AccordionContent;
Accordion.ContentDetails = AccordionContentDetails;
Accordion.ContentFooter = AccordionContentFooter;
Accordion.Header = AccordionHeader;
export default Accordion;
