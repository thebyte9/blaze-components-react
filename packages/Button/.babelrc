{
  "presets": ["@babel/preset-env", "@babel/preset-react", "@babel/preset-typescript"],
  "plugins": [
    [
      "@babel/plugin-transform-runtime",
      {
        "regenerator": true
      }
    ],
    "@babel/plugin-syntax-jsx",
    "@babel/proposal-class-properties",
    "@babel/proposal-object-rest-spread"
  ]
}
