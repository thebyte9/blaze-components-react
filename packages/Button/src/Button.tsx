import React from 'react';
import { buildClassNames } from '@blaze-react/utils';

type TType = 'button' | 'submit' | 'reset';
export interface IButtonProps {
  disabled?: boolean;
  children?: unknown;
  utilities?: string;
  modifiers?: string[];
  type?: TType;
}

export const Button = ({ children, utilities = '', ...rest }: IButtonProps): JSX.Element => {
  const { modifiers = [], disabled, type = 'button' } = rest;

  const DeprecatedButton = () => {
    console.warn('Modifiers will be deprecated in the near future. You should use CSS utilities classes instead');

    const formatedModifiers: string = modifiers.map((modifier) => `button--${modifier}`).join(' ');
    const buttonClassNames: string = buildClassNames('button', {
      [formatedModifiers]: !!modifiers,
    });

    return (
      <button disabled={disabled} className={buttonClassNames} type={type} {...rest}>
        {children}
      </button>
    );
  };

  const BlazeButton = (
    <button type="button" className={utilities} {...rest}>
      {children}
    </button>
  );

  return utilities !== '' ? BlazeButton : DeprecatedButton();
};
