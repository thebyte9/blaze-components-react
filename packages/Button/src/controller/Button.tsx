import React from 'react';
import { DisplayIconProps, IButtonProps } from '../types';
import { ButtonView } from '../view/ButtonView';
import { Icon } from '@blaze-react/icon';
import { useComponentLogic } from '../hooks/useComponentLogic';

export const Button = ({
  label,
  type,
  variant,
  iconOnly = false,
  disabled = false,
  icon,
  displayIcon = DisplayIconProps.NoIcon,
  overrides = [],
  children,
  theme,
  onClick,
}: IButtonProps): JSX.Element => {
  const { display, classes } = useComponentLogic({
    componentVariant: variant,
    theme: theme,
    displayIcon,
    iconOnly: iconOnly,
    overrides: overrides,
  });

  return (
    <ButtonView classes={classes} type={type} disabled={disabled} onClick={onClick}>
      <Icon display={display} iconOnly={iconOnly} label={label} icon={icon} />
      {children}
    </ButtonView>
  );
};
