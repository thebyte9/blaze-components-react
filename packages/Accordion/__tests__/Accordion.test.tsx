import Button from "../../Button/src/Button";
import React from "react";
import Accordion from "../src/Accordion";
import { render, fireEvent } from '@testing-library/react'

const AccordionComponent = (
  <Accordion>
    <Accordion.Header>
      <p>Accordion text 1</p>
      <p>Accordion text 2</p>
    </Accordion.Header>
    <Accordion.Content>
      <Accordion.ContentDetails>
        <p>Content here. Components can be placed in here if needed</p>
        <p>Footer can be added below if needed</p>
      </Accordion.ContentDetails>
      <Accordion.ContentFooter>
        <Button name="button" modifiers={["outline", "light", "rounded"]}>
          Cancel
        </Button>
        <Button name="button" modifiers={["rounded"]}>
          Save
        </Button>
      </Accordion.ContentFooter>
    </Accordion.Content>
  </Accordion>
);

describe("Accordion component", () => {
  it('should match snapshot', () => {
    const { asFragment } = render(AccordionComponent);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should toggle on click', () => {
    const wrapper = render(AccordionComponent);
    const toggleAccordion = wrapper.getByTestId('toggle-accordion')
    fireEvent.click(toggleAccordion);
  })
});
