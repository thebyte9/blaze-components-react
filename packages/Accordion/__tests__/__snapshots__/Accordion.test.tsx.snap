// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component should be defined and renders correctly (snapshot) 1`] = `
<Accordion>
  <div
    className="accordion"
  >
    <div
      className="accordion__header"
    >
      <AccordionHeader>
        <div
          className="accordion__header-text"
        >
          <p>
            Accordion text 1
          </p>
          <p>
            Accordion text 2
          </p>
        </div>
      </AccordionHeader>
      <div
        className="icon-button icon-button--accordion"
        onClick={[Function]}
      >
        <i
          className="material-icons"
        >
          keyboard_arrow_down
        </i>
      </div>
    </div>
    <div
      className="accordion__content-wrapper"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <AccordionContent>
        <AccordionContentDetails>
          <div
            className="accordion__content"
          >
            <p>
              Content here. Components can be placed in here if needed
            </p>
            <p>
              Footer can be added below if needed
            </p>
          </div>
        </AccordionContentDetails>
        <AccordionContentFooter>
          <div
            className="accordion__footer"
          >
            <withUtils()
              disabled={false}
              modifiers={
                Array [
                  "outline",
                  "light",
                  "rounded",
                ]
              }
              name="button"
              type="button"
            >
              <Component
                disabled={false}
                modifiers={
                  Array [
                    "outline",
                    "light",
                    "rounded",
                  ]
                }
                name="button"
                type="button"
                utils={
                  Object {
                    "ErrorMessage": [Function],
                    "buildClassNames": [Function],
                    "removeExtraSpaces": [Function],
                    "uniqueId": [Function],
                  }
                }
              >
                <button
                  className="button button--outline button--light button--rounded"
                  disabled={false}
                  name="button"
                  type="button"
                >
                  Cancel
                </button>
              </Component>
            </withUtils()>
            <withUtils()
              disabled={false}
              modifiers={
                Array [
                  "rounded",
                ]
              }
              name="button"
              type="button"
            >
              <Component
                disabled={false}
                modifiers={
                  Array [
                    "rounded",
                  ]
                }
                name="button"
                type="button"
                utils={
                  Object {
                    "ErrorMessage": [Function],
                    "buildClassNames": [Function],
                    "removeExtraSpaces": [Function],
                    "uniqueId": [Function],
                  }
                }
              >
                <button
                  className="button button--rounded"
                  disabled={false}
                  name="button"
                  type="button"
                >
                  Save
                </button>
              </Component>
            </withUtils()>
          </div>
        </AccordionContentFooter>
      </AccordionContent>
    </div>
  </div>
</Accordion>
`;
